synapse/lib/view.py

```diff
@@ -824,6 +824,33 @@ async def getPropArrayCount(self, propname, valu=s_common.novalu):
 
         return count
 
+    async def iterPropUniqValues(self, propname):
+        prop = self.core.model.prop(propname)
+        if prop is None:
+            mesg = f'No property named {propname}'
+            raise s_exc.NoSuchProp(mesg=mesg)
+
+        formname = None
+        propname = None
+
+        if prop.isform:
+            formname = prop.name
+        else:
+            propname = prop.name
+            if not prop.isuniv:
+                formname = prop.form.name
+
+        genrs = [layr.iterPropUniqValues(formname, propname, prop.type.stortype) for layr in self.layers]
+
+        lastvalu = None
+        async for indx, valu in s_common.merggenr2(genrs):
```

Comment:
Maybe this was already discussed, but it feels counter-intuitive to me that this would yield values from lower layers that you would not lift from the view.  e.g.
```
[ media:news=(foo,) :title=foobar ] // default view
[ media:news=(foo,) :title=barbam ] // in a fork
This api in the fork returns both foobar and barbam.
```

Reply:
Yeah I'll tweak it to prevent that, probably a bit slower but might not be too bad.

Reply:
updated to verify the prop exists in the view